SonarQube.Common/ProcessRunner.cs

```diff
@@ -64,6 +87,16 @@ public bool Execute(string exeName, string args, string workingDirectory, int ti
                 WorkingDirectory = workingDirectory
             };
 
+            if (envVariables != null)
+            {
+                foreach (KeyValuePair<string, string> envVariable in envVariables)
+                {
+                    if (!String.IsNullOrEmpty(envVariable.Value))
+                    {
+                        psi.EnvironmentVariables.Add(envVariable.Key, envVariable.Value);
```

Comment:
Might be worth adding "child processes do not automatically inherit..." as a doc comment, either as a <remark> or in the documentation for the envVariables parameter